tt_metal/llrt/tt_cluster.cpp

```diff
@@ -96,6 +98,11 @@ void Cluster::detect_arch_and_target() {
         this->target_type_);
 }
 
+void Cluster::initialize_hal() {
```

Comment:
right now this initializes the HAL and also populates `dram_address_params`, can we remove this and just call `tt_metal::hal.initialize(arch_)` from the caller. 

Also we don't need to hold onto `dram_address_params`, I think it would be okay to remove this from tt_cluster and just pass in `dram_address_params` into the UMD constructor

Reply:
Good ideas! Implemented in commit https://github.com/tenstorrent/tt-metal/pull/13510/commits/ae39f4545dd46926592b4752d8898753e98cd485